tests/mochainon.d.ts

```diff
@@ -0,0 +1,12 @@
+declare module 'mochainon';
+
+import * as chai from 'chai';
+import * as sinon from 'sinon';
+
+interface Mochainon {
+	chai: typeof chai;
+	sinon: typeof sinon;
+}
+
+declare const mochainon: Mochainon;
+export default mochainon;
```

Comment:
Ok: this isn't a type definition, it's an actual module, which is why it's not successfully defining 'mochainon' for your test file. If you have an import or an export at the top-level, you're a module, and modules aren't allowed to do `declare` statements at all (I'm not sure why it isn't in the compiler output with the other errors, but for me in VSC I get an error on the declare at the top here).

You don't want this. You want a module declaration: to do that, add a body to your `declare module 'mochainon'` statement, that defines the contents of that module. That's pretty close to what you've got here (but with braces around the whole lot).

There's on other bug here: you've used `export default`, but it's not actually a default export (`require('mochainon').default` doesn't exist) - it should be `export = mochainon`.